.github/PULL_REQUEST_TEMPLATE.md

```diff
@@ -0,0 +1,24 @@
+### Description
+
+_Please explain the changes you made here._
+_Explain the **motivation** for making this change. What existing problem does the pull request solve?_
+_Mention any issues fixed, addressed or otherwise related to this pull request, including issue numbers or hard links for issues in other repos._
+
+----
+
+### Checklist (never delete this)
+
+Never delete this, it is our record that procedure was followed. If you find that for whatever reason one of the checklist points doesn't apply to your PR, you can leave it unchecked but please add an explanation below. 
+
+#### Content
+- [ ] Added or modified tests to cover changes and any new functionality
+- [ ] All tests passing on Travis 
+
+#### Form
+- [ ] Rebased, squashed and reworded to produce a single commit (exemptions may be made -- sparingly) that is _uncluttered by excess lines_ (lines that just say "addressed review", "fixed" etc left over from squashing)
```

Comment:
I think we should a link to a wiki page that describes how to write a good commit message.  It's so much easier when searching history.